WordPress/gutenberg
# Swatch: Convert component to TypeScript

## What?
Refactor the `Swatch` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored the component TypeScript

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test the various use of Swatch within the block/site editors in Gutenberg still work
